controller/src/beerocks/master/db/db.cpp

```diff
@@ -2737,6 +2737,21 @@ bool db::clear_client_persistent_db(const sMacAddr &mac)
     return true;
 }
 
+bool db::is_hostap_on_client_selected_bands(const int8_t selected_bands, const sMacAddr &hostap)
+{
+    auto hostap_band = wireless_utils::which_freq(get_node_channel(tlvf::mac_to_string(hostap)));
+
+    switch (hostap_band) {
+    case beerocks::eFreqType::FREQ_24G:
+        return (selected_bands & eClientSelectedBands::eSelectedBands_24G);
+    case beerocks::eFreqType::FREQ_5G:
+        return (selected_bands & eClientSelectedBands::eSelectedBands_5G);
+    default:
+        LOG(WARNING) << "hostap band " << hostap_band << "  is not supported by client";
+        return false;
+    }
+}
+
```

Comment:
i does access get_node_channel() .
but now that i think of i should probably change the signature to 
`bool db::is_hostap_on_client_selected_bands(const sMacAddr client, const sMacAddr &hostap)`

Reply:
So what it access to `get_node_channel()`? It is a public database function and can be accessed from outside the database.
You can do a global search and see that it used on many places from outside the database.
just instead of call it directly, you would have to add `database.` before: `database.get_node_channel()`.
I really believe that the database is not the right place for this function and that.

Reply:
i changed the signature to  ` db::is_hostap_on_client_selected_bands(const sMacAddr client, const sMacAddr &hostap)`
since this function can be uses from other tasks it belongs to the db (for example client_association_task )